### PR TITLE
Restructure pages

### PR DESCRIPTION
**Pull request by Valioesi**
I had do solve a few more issues and redefined the meaning of the showForOwnMunicipality flag in contentful a bit. You can now define if a section should be shown only for one's own municipality, explicitly only for everything except one's own municipality, or just everywhere (see picture).

I tested the pages by comparing them to production site: startpage signed out, startpage signed in, municipality page signed out, own municipality page signed in, other municipality page signed in. 

Everything seemed fine after some adjustments, but maybe go through it as well. :) 

grafik

### CODE REVIEW

File: src/components/Layout/Sections/RenderPage.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    userData,
+    isAuthenticated,
+    sections,
+    pageContext,
+    userMunicipalityState,
+  })?.map((section, index) => (
+    <React.Fragment key={index}>{section}</React.Fragment>
+  ));
+
+  return <SectionWrapper>{renderedSections}</SectionWrapper>;
```
Comment (nmauersberg) [inline src/components/Layout/Sections/RenderPage.js:24]: Much cleaner this way! :)

File: src/hooks/Municipality/UserMunicipalityState/index.js [hunk 2]
```diff
@@ -2,33 +2,33 @@ import { useContext, useEffect, useState } from 'react';
...
     } else {
-      setUserContentfulState('loggedOut');
+      setUserMunicipalityState('loggedOut');
     }
   }, [isAuthenticated, customUserData, municipality]);
 
   // Debug Contentful showForOptions with this log:
   // console.log(
-  //   `*\nContentful state:\nmunicipality: ${municipalityContentfulState}, user: ${userContentfulState}\n\n`
+  //   `*\nContentful state:\nmunicipality: ${municipalityContentfulState}, user: ${userMunicipalityState}\n\n`
```
Comment (nmauersberg) [inline src/hooks/Municipality/UserMunicipalityState/index.js:30]: It's just a debug log, but "municipalityContentfulState" seems not defined.

File: src/components/utils/index.js [hunk 3]
```diff
@@ -252,73 +252,58 @@ export function mapCampaignCodeToAgs(campaignCode) {
...
+    // municipality based on the ags attribute in contentful. If there is a default section
+    // with a specific sectionId it should be overwritten, if there is specific section with an ags.
+    if (municipality?.ags) {
+      if (el.ags?.length) {
+        if (!el.ags.includes(municipality.ags)) {
+          showState = false;
         }
-      }
-
-      if (
```
Comment (nmauersberg) [inline src/components/utils/index.js:310]: Ah, was this the reason, why some components rendered multiple times?

**Review by nmauersberg — APPROVED** [review]
For me it worked as the production site too. While going through I've seen that on the logged in municipality page there were several white background components in a row. Everything else looked fine.

File: src/hooks/Municipality/UserMunicipalityState/index.js (hunk 2 above)
Comment (Valioesi) [reply src/hooks/Municipality/UserMunicipalityState/index.js:30]: Ah yeah, I renamed it. Going to change it.

File: src/components/utils/index.js (hunk 3 above)
Comment (Valioesi) [reply src/components/utils/index.js:310]: Maybe...can't remember anymore. :D

**Pull request merged**